solidity-v1/dashboard/src/components/modal/threshold/WithdrawGrantedTokens.jsx

```diff
@@ -0,0 +1,195 @@
+      <ModalBody>
+        <h3 className={"mb-1"}>Withdraw granted tokens</h3>
+        <p className={"mb-2"}>
+          The following tokens are available to be released and withdrawn from
+          your token grant.
+        </p>
+        <div>
+          <div className="withdraw-granted-tokens__info-row">
+            <h4 className="withdraw-granted-tokens__info-row-title">
+              Withdraw:
+            </h4>
+            <span className="withdraw-granted-tokens__info-row-value">
+              total:{" "}
+              <TokenAmount
+                amount={totalReadyToRelease}
+                token={KEEP}
+                wrapperClassName="withdraw-granted-tokens__total-ready-to-release-amount"
+                amountClassName="text-grey-60"
+                symbolClassName="text-grey-60"
+              />
+            </span>
+          </div>
+          <form>
+            {grants
+              .slice(0, numberOfGrantsDisplayed)
+              .map((grant) => renderGrant(grant, grants.length, formik))}
+          </form>
+          <OnlyIf condition={grants.length > numberOfGrantsDisplayed}>
+            <div
+              className="withdraw-granted-tokens__view-more-container"
+              onClick={() =>
+                setNumberOfGrantsDisplayed(numberOfGrantsDisplayed + 5)
+              }
+            >
+              <span>+&nbsp;</span>
+              View More
+            </div>
+          </OnlyIf>
+        </div>
+      </ModalBody>
+      <ModalFooter>
+        <SubmitButton
+          className="btn btn-primary btn-lg mr-2"
+          type="submit"
+          onSubmitAction={(awaitingPromise) => {
+            onWithdrawClick(awaitingPromise)
+          }}
+          disabled={formik.values.grantIds.length === 0}
+        >
+          withdraw
+        </SubmitButton>
+        <Button className={`btn btn-unstyled text-link`} onClick={onClose}>
+          Cancel
+        </Button>
+      </ModalFooter>
+    </>
+  )
+})
+
+const renderGrant = (grant, totalNumberOfGrants, formik) => {
+  return (
+    <div
+      key={`Grant-${grant.id}`}
+      style={{
+        display: "flex",
+        alignItems: "baseline",
+      }}
+    >
+      <FormCheckboxBase
+        className="withdraw-granted-tokens__form_checkbox mr-1"
+        name="grantIds"
+        type="checkbox"
+        value={grant.id}
+        checked={formik.values.grantIds.includes(grant.id)}
+        onChange={(e) => {
+          const set = new Set(formik.values.grantIds)
+          if (e.target.checked) {
+            set.add(grant.id)
+          } else {
+            set.delete(grant.id)
+          }
+          formik.setFieldValue("grantIds", [...set])
+        }}
+      />
+      <Accordion
+        allowZeroExpanded={totalNumberOfGrants > 1}
+        className={"withdraw-granted-tokens__grants-accordion"}
+      >
+        <AccordionItem
+          {...(totalNumberOfGrants === 1
+            ? { dangerouslySetExpanded: true }
+            : {})}
```

Comment:
We actually can't because when we put `dangerouslySetExpanded` prop (no matter what value) then the items cannot be expanded.

The current way is we create the `dangerouslySetExpanded` only when the total number of grants is equal 1, and if not then we don't put this prop at all

Reply:
Also I realized that we render an `Acordion` wrapper for every grant but I belive we should render the `AccordionItem` per grant. So in the parent component we should have:
```
<Accordion preExpanded={grants.length > 1 ? [] : [grants[0].id } {...restprops} >
 {grants.slice(...).map(renderGrant)}
</Accordion>
```

and the `GrantItem` component:
```
<AccordionItem uuid={grant.id}>
 {....}
</AccordionItem>
```

Reply:
> What about [`preExpanded` prop](https://github.com/springload/react-accessible-accordion#preexpanded-string-optional-default-)?

Fixed in 41d03daeb2b777e292a29a218c271ea33c021e83

Reply:
> Also I realized that we render an `Acordion` wrapper for every grant but I belive we should render the `AccordionItem` per grant. So in the parent component we should have:
> 
> ```
> <Accordion preExpanded={grants.length > 1 ? [] : [grants[0].id } {...restprops} >
>  {grants.slice(...).map(renderGrant)}
> </Accordion>
> ```
> 
> and the `GrantItem` component:
> 
> ```
> <AccordionItem uuid={grant.id}>
>  {....}
> </AccordionItem>
> ```

Fixed in 9369c4b949905a6415407064fb6225228c6cee7d and c9aa80336634f71e4b2c777d5b1fe76230591b40